compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCastGen.java

```diff
@@ -115,35 +115,26 @@ public static void generateBToJCheckCast(MethodVisitor mv, BType sourceType, JTy
 
         if (targetType.jTag == JTypeTags.JBYTE) {
```

Comment:
I think we can use switch statement here.

Reply:
yes, we need to use. waiting for the BString migration to complete.